ui/src/main/webapp/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -13,15 +14,27 @@ import {
 import CheckboxSubGroup from './CheckboxSubGroup';
 import CheckboxRowWrapper from './CheckboxRowWrapper';
 import { useValidation } from './checkboxGroupValidation';
+import { MODE_CREATE } from '../../constants/modes';
 
 function CheckboxGroup(props: CheckboxGroupProps) {
-    const { field, value, handleChange, controlOptions, addCustomValidator } = props;
+    const { field, handleChange, controlOptions, addCustomValidator } = props;
 
     const flattenedRowsWithGroups = getFlattenRowsWithGroups(controlOptions);
+    const value =
+        props.mode === MODE_CREATE
+            ? getDefaultValues(controlOptions.rows)
+            : parseValue(props.value);
 
-    const [values, setValues] = useState(parseValue(value));
-    useValidation(addCustomValidator, field, controlOptions);
+    // propagate defaults up if the component is not touched
+    useEffect(() => {
+        if (props.mode === MODE_CREATE) {
```

Comment:
is there a possibility for that component to be used as a single input in separated tab?
if yes then the mode can be `MODE_CONFIG= 'config`
it is due to single element displayed on tab is used as config instead of create,

![Screenshot 2023-10-13 at 11 22 42](https://github.com/splunk/addonfactory-ucc-generator/assets/143183665/2ffbb2bd-b77c-485d-8a89-146a3555c338)

on this image the mode is propagated with config even when it might look as just create, so if you check it here i assume it might cause some issues

Reply:
I don't expect there 'config' because it is not allowed by schema.json to use in in configuration (which may be changed in the future)
But yes, it is totally fine to use it as standalone component. In this case, the defaulValue for checkbox will be ignored.